pkg/scanners/scannerv4/nodescan_convert.go

```diff
@@ -0,0 +1,152 @@
+package scannerv4
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+var (
+	rhcosOSImageRegexp = regexp.MustCompile(`(Red Hat Enterprise Linux) (CoreOS) ([\d])([\d]+)`)
+)
+
+const (
+	rhcosFullName = "Red Hat Enterprise Linux CoreOS"
+)
+
+func toNodeScan(r *v4.VulnerabilityReport, osImageRef string) *storage.NodeScan {
+	// TODO(ROX-26593): Instead of fixing notes here, add RHCOS DistributionScanner to ClairCore
+	fixedNotes := fixNotes(toStorageNotes(r.Notes), osImageRef)
+
+	convertedOS := toOperatingSystem(osImageRef)
+	if convertedOS == "" {
+		log.Warnf("Could not determine operating system from OSimage ref %s", osImageRef)
+	}
+
+	return &storage.NodeScan{
+		ScanTime:        protocompat.TimestampNow(),
+		Components:      toStorageComponents(r),
```

Comment:
how much does this differ from `components` in `convert.go`? Can we just reuse that instead?